File: GopherVerse/Episode1/primecount/primecount.go
package main

import (
	"fmt"
	"math"
	"time"
)

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	sqrt := int(math.Sqrt(float64(n)))
	for i := 2; i <= sqrt; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	start := time.Now()
	count := 0
	for i := 2; i <= 10000000; i++ {
		if isPrime(i) {
			count++
		}
	}
	elapsed := time.Since(start).Milliseconds()
	fmt.Printf("Total primes: %d\n", count)
	fmt.Printf("Execution time: %d ms\n", elapsed)
}

File: GopherVerse/Episode1/primecount/primecount.py
import math
import time

def is_prime(n):
    if n <= 1:
        return False
    sqrt_n = int(math.sqrt(n))
    for i in range(2, sqrt_n + 1):
        if n % i == 0:
            return False
    return True

start = time.perf_counter()
count = 0
for i in range(2, 10_000_001):
    if is_prime(i):
        count += 1
elapsed = (time.perf_counter() - start) * 1000
print(f"Total primes: {count}")
print(f"Execution time: {elapsed:.2f} ms")
